### ISSUE TITLE
complete node

### ISSUE THREAD

**Issue by garrathbond** [bartbutenaers/node-red-contrib-msg-resend#12]
Cant seem to get this to work with the "complete" node Should it?

**Comment by bartbutenaers** [comment]
@garrathbond,

Most of my nodes have been written before the whole Complete mechanism was introduced in Node-RED.   So it woyld mean a lot of time for me to implement this.  Moreover I had the expectation that most nodes wouldn't do it, for the same reason.  And I think I am right, because you are the first one asking me, about any of my nodes.

This kind of stuff should have been introduced much much earlier in Node-RED, especially before more than thousand of nodes already existed.

Barr

**Comment by garrathbond** [comment]
ok thanks anyway good to know I wasnt missing something

**Comment by bartbutenaers** [comment]
@garrathbond,

I had a quick look at the done-mechanism, introduced [here](https://nodered.org/blog/2019/09/20/node-done) at the time being.  I have tried to implement it, although it is not 100% clear to me how it should behave in all case, e.g. when errors and so on...  Especially since I have never used the Complete node myself.

I tried this example flow, which let's me know that the msg is done as soon as it has been resend 5 times:

![image](https://user-images.githubusercontent.com/14224149/229912969-05b1f6cf-7ad0-478e-9865-2773f05d9b5a.png)

Is that the behaviour you had expected?

Would be nice if you could do some tests, also with incorrect inputs.  To see if it behaves correctly.  Then I will publish it afterwards on npm.

You can install it like this from my Github account, from within your .node-red folder (assuming you have git installed):
```
npm install bartbutenaers/node-red-contrib-msg-resend
```

Bart

**Comment by garrathbond** [comment]
Sorry I only install from the node red palette(in side “home assistant”). (I don’t really understand unix based system more windows based) and your node is not showing any update!

**Comment by bartbutenaers** [comment]
Yes it is normal that no updates are showing.  As explained I above I had hoped that you wanted to test my changes, and only afterwards I would publish the changes on npm (and only then they become available in the Node-RED palette).

**Comment by garrathbond** [comment]
Yes happy to try but not very good with unix so don’t really know how to install without messing thing up(sorry)
